cluster/manager_delete.go

```diff
@@ -56,38 +54,177 @@ func (m *Manager) DeleteCluster(ctx context.Context, cluster CommonCluster, forc
 	return nil
 }
 
-func deleteAllResource(kubeConfig []byte, logger *logrus.Entry) error {
+func retry(function func() error, count int, delaySeconds int) error {
```

Comment:
This function should probably be factored out into a utility package.

Reply:
Definitely not. Utility packages are evil. Until we have proper test coverage talking about code reusability makes no sense at all.

BUT: I would rather say, that we should probably use some sort of backoff library: https://medium.com/@lestrrat/yak-shaving-with-backoff-libraries-in-go-80240f0aa30c

Ultimately this code go into a pkg/backoff package itself, but there are already great implementations for that.

Reply:
we can always move this to a utility package, if it can be used at least in another source file. we could also add an external dependency, but the boilerplate to use any of them is longer than this function alone. i would not change this for now.